From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alistair Delva <adelva@google.com>
Date: Thu, 8 Apr 2021 09:50:06 -0700
Subject: ANDROID: rockpi4: Update with probed drivers

Optimize the drivers list for those that are defined in DT, but also
probed on a real device. Sort the modules into groups so we know why
they are pulled in.

Bug: 183963484
Signed-off-by: Alistair Delva <adelva@google.com>
Change-Id: I6edef7df561b46154857ba21424a925eec0e8acf
---
 arch/arm64/configs/rockpi4_gki.fragment | 97 +++++++++++++++----------
 1 file changed, 59 insertions(+), 38 deletions(-)

diff --git a/arch/arm64/configs/rockpi4_gki.fragment b/arch/arm64/configs/rockpi4_gki.fragment
--- a/arch/arm64/configs/rockpi4_gki.fragment
+++ b/arch/arm64/configs/rockpi4_gki.fragment
@@ -1,59 +1,80 @@
+# Core features
 CONFIG_ARCH_ROCKCHIP=y
-CONFIG_DEVTMPFS=y
-CONFIG_ZRAM=m
-CONFIG_ZSMALLOC=m
+# CONFIG_CLK_PX30 is not set
+# CONFIG_CLK_RV110X is not set
+# CONFIG_CLK_RK3036 is not set
+# CONFIG_CLK_RK312X is not set
+# CONFIG_CLK_RK3188 is not set
+# CONFIG_CLK_RK322X is not set
+# CONFIG_CLK_RK3288 is not set
+# CONFIG_CLK_RK3308 is not set
+# CONFIG_CLK_RK3328 is not set
+# CONFIG_CLK_RK3368 is not set
+CONFIG_COMMON_CLK_RK808=m
+CONFIG_CPUFREQ_DT=m
+CONFIG_MFD_RK808=m
+CONFIG_PCIE_ROCKCHIP_HOST=m
+CONFIG_PHY_ROCKCHIP_PCIE=m
+CONFIG_PL330_DMA=m
+CONFIG_PWM_ROCKCHIP=m
+CONFIG_PWRSEQ_SIMPLE=m
+CONFIG_REGULATOR_FAN53555=m
+CONFIG_REGULATOR_PWM=m
+CONFIG_REGULATOR_RK808=m
+CONFIG_ROCKCHIP_EFUSE=m
+CONFIG_ROCKCHIP_IOMMU=y
+CONFIG_ROCKCHIP_IODOMAIN=m
+CONFIG_ROCKCHIP_MBOX=y
+CONFIG_ROCKCHIP_PM_DOMAINS=y
+CONFIG_ROCKCHIP_THERMAL=m
+
+# Ethernet
 CONFIG_STMMAC_ETH=m
 # CONFIG_DWMAC_GENERIC is not set
 # CONFIG_DWMAC_IPQ806X is not set
 # CONFIG_DWMAC_QCOM_ETHQOS is not set
 # CONFIG_DWMAC_SUNXI is not set
 # CONFIG_DWMAC_SUN8I is not set
-CONFIG_ROCKCHIP_PHY=m
-CONFIG_I2C_DESIGNWARE_SLAVE=y
-CONFIG_I2C_DESIGNWARE_PLATFORM=m
+
+# I2C
 CONFIG_I2C_RK3X=m
-CONFIG_SPI_ROCKCHIP=m
-CONFIG_PINCTRL_RK805=m
-CONFIG_SYSCON_REBOOT_MODE=m
-CONFIG_ROCKCHIP_THERMAL=m
+
+# Watchdog
 CONFIG_DW_WATCHDOG=m
-CONFIG_MFD_RK808=m
-CONFIG_REGULATOR_RK808=m
+
+# Display
 CONFIG_DRM_ROCKCHIP=m
+CONFIG_ROCKCHIP_ANALOGIX_DP=y
 CONFIG_ROCKCHIP_DW_HDMI=y
-CONFIG_SND_SOC_TS3A227E=m
+CONFIG_ROCKCHIP_DW_MIPI_DSI=y
+
+# USB 2.x
+CONFIG_PHY_ROCKCHIP_INNO_USB2=m
 CONFIG_USB_OHCI_HCD=m
 # CONFIG_USB_OHCI_HCD_PCI is not set
 CONFIG_USB_OHCI_HCD_PLATFORM=m
+
+# eMMC / SD-Card
 CONFIG_MMC_SDHCI_OF_ARASAN=m
-CONFIG_MMC_SDHCI_OF_DWCMSHC=m
 CONFIG_MMC_DW=m
 CONFIG_MMC_DW_ROCKCHIP=m
-CONFIG_RTC_DRV_RK808=m
-CONFIG_PL330_DMA=m
-# CONFIG_CLK_PX30 is not set
-# CONFIG_CLK_RV110X is not set
-# CONFIG_CLK_RK3036 is not set
-# CONFIG_CLK_RK312X is not set
-# CONFIG_CLK_RK3188 is not set
-# CONFIG_CLK_RK322X is not set
-# CONFIG_CLK_RK3288 is not set
-# CONFIG_CLK_RK3308 is not set
-# CONFIG_CLK_RK3328 is not set
-# CONFIG_CLK_RK3368 is not set
-CONFIG_COMMON_CLK_RK808=m
-CONFIG_ROCKCHIP_MBOX=y
-CONFIG_ROCKCHIP_IOMMU=y
-CONFIG_ROCKCHIP_IODOMAIN=m
-CONFIG_ROCKCHIP_PM_DOMAINS=y
-CONFIG_ARM_RK3399_DMC_DEVFREQ=m
-CONFIG_PWM_ROCKCHIP=m
 CONFIG_PHY_ROCKCHIP_EMMC=m
-CONFIG_PHY_ROCKCHIP_INNO_HDMI=m
-CONFIG_PHY_ROCKCHIP_INNO_USB2=m
-CONFIG_PHY_ROCKCHIP_USB=m
-CONFIG_ROCKCHIP_EFUSE=m
-CONFIG_CRYPTO_DEV_ROCKCHIP=m
+
+# Real-time clock
+CONFIG_RTC_DRV_RK808=m
+
+# Type-C
+CONFIG_PHY_ROCKCHIP_TYPEC=m
+
+# SAR ADC
+CONFIG_ROCKCHIP_SARADC=m
+
+# Audio
+CONFIG_SND_SOC_ROCKCHIP_I2S=m
+
+# To boot Linux distributions like Debian
+CONFIG_DEVTMPFS=y
+
 # To bootstrap rootfs with QEMU
 CONFIG_VIRTIO_PCI=m
 CONFIG_VIRTIO_BLK=m
